File: FrontEnd/FinancasApp/services/Transacao.ts
import {api} from "./api"

export type Transacao = {
    id: number,
    descricao: string,
    valor: number,
    tipo: 'Despesa' | 'Receita',
    categoria: string,
    pessoaId: number
}

export async function getTransacoes(): Promise<Transacao[]> {
    return api<Transacao[]>('transacao', 'GET');
}

export async function postTransacao(transacao: Transacao): Promise<Transacao> {
    return api<Transacao>('transacao', 'POST', transacao);
}

File: BackEnd/FinancasApi/FinancasApi/Models/Transacao.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FinancasApi.Models
{
    [Table("Transacao")]
    public class Transacao
    {
        [Key]
        public int Id { get; set; }

        [Required(ErrorMessage = "O campo 'Descrição' é obrigatório.")]
        [StringLength(400, ErrorMessage = "A descrição deve ter no máximo 400 caracteres.")]
        public string Descricao { get; set; } = string.Empty;

        [Required(ErrorMessage = "O campo 'Valor' é obrigatório.")]
        [Range(0.01, double.MaxValue, ErrorMessage = "O valor deve ser maior que zero.")]
        [Column(TypeName = "NUMERIC")]
        public decimal Valor { get; set; }

        [Required(ErrorMessage = "O campo 'Tipo' é obrigatório.")]
        public TipoTransacao Tipo { get; set; }

        [Required(ErrorMessage = "O campo 'Categoria' é obrigatório.")]
        public string Categoria { get; set; } = string.Empty;

        [Required(ErrorMessage = "O campo 'PessoaId' é obrigatório.")]
        public int PessoaId { get; set; }
    }

    public enum TipoTransacao
    {
        Receita,
        Despesa
    }
}
